test/runtime/Policies.go

```diff
@@ -425,6 +426,52 @@ var _ = Describe("RuntimePolicies", func() {
 		}
 	})
 
+	It("Checks that traffic is not dropped when L4 policy is installed and deleted", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		srvIP, err := vm.ContainerInspectNet(helpers.Httpd1)
+		Expect(err).Should(BeNil(), "Cannot get httpd1 server address")
+		type BackgroundTestAsserts struct {
+			res  *helpers.CmdRes
+			time time.Time
+		}
+		backgroundChecks := []*BackgroundTestAsserts{}
+		go func() {
+			for {
+				select {
+				default:
+					res := vm.ContainerExec(
+						helpers.App1,
+						helpers.CurlFail("http://%s/", srvIP[helpers.IPv4]))
+					assert := &BackgroundTestAsserts{
+						res:  res,
+						time: time.Now(),
+					}
+					backgroundChecks = append(backgroundChecks, assert)
+				case <-ctx.Done():
+					return
+				}
+			}
+		}()
+		// Sleep a bit to make sure that the goroutine starts.
+		time.Sleep(50 * time.Millisecond)
+
+		_, err = vm.PolicyImportAndWait(vm.GetFullPath(policiesL4Json), helpers.HelperTimeout)
+		Expect(err).Should(BeNil(), "Cannot install L4 policy")
+
+		By("Uninstalling policy")
+		vm.PolicyDelAll().ExpectSuccess("Cannot delete all policies")
+		vm.WaitEndpointsReady()
+
+		By("Canceling background connections from app2 to httpd1")
+		cancel()
```

Comment:
This will not do what you will be expecting, when calling `cancel()` it doesn't mean that the go routine created in line 439 is completed. Add a sync.WaitGroup in line 451 with `wg.Done()` and do `wg.Wait()` here.

Reply:
I mean, need to cancel the function, If a  Waiting group is added, when you stop the background loop? Is never done

And I'm making sure that the function works, we check that the `backgroundChecks` are not empty.

`Expect(backgroundChecks).ShouldNot(BeEmpty(), "No background connections were made")`

If we use a WG, can be waiting without timeout, and block the operation until jenkins timeout, I'm not  sure If I get what you propose :-(

Reply:
```golang
		go func() {
			for {
				select {
				default:
					res := vm.ContainerExec(
						helpers.App1,
						helpers.CurlFail("http://%s/", srvIP[helpers.IPv4]))
					assert := &BackgroundTestAsserts{
						res:  res,
						time: time.Now(),
					}
					backgroundChecks = append(backgroundChecks, assert)
				case <-ctx.Done():
					return
				}
			}
		}()
		// Sleep a bit to make sure that the goroutine starts.
		time.Sleep(50 * time.Millisecond)
 		_, err = vm.PolicyImportAndWait(vm.GetFullPath(policiesL4Json), helpers.HelperTimeout)
This conversation was marked as resolved by eloycoto
 		Expect(err).Should(BeNil(), "Cannot install L4 policy")
 		By("Uninstalling policy")
		vm.PolicyDelAll().ExpectSuccess("Cannot delete all policies")
		vm.WaitEndpointsReady()
 		By("Canceling background connections from app2 to httpd1")
		cancel()
		GinkgoPrint("Made %d connections in total", len(backgroundChecks))
```
When you call "cancel()" the `backgroundChecks` can still be written and the `len(backgroundChecks)` might be something that we are not expecting. Do something like this:


```golang
		go func() {
			for {
				select {
				default:
					res := vm.ContainerExec(
						helpers.App1,
						helpers.CurlFail("http://%s/", srvIP[helpers.IPv4]))
					assert := &BackgroundTestAsserts{
						res:  res,
						time: time.Now(),
					}
					backgroundChecks = append(backgroundChecks, assert)
				case <-ctx.Done():
+				       wg.Done()
					return
				}
			}
		}()
		// Sleep a bit to make sure that the goroutine starts.
		time.Sleep(50 * time.Millisecond)
 		_, err = vm.PolicyImportAndWait(vm.GetFullPath(policiesL4Json), helpers.HelperTimeout)
This conversation was marked as resolved by eloycoto
 		Expect(err).Should(BeNil(), "Cannot install L4 policy")
 		By("Uninstalling policy")
		vm.PolicyDelAll().ExpectSuccess("Cannot delete all policies")
		vm.WaitEndpointsReady()
 		By("Canceling background connections from app2 to httpd1")
		cancel()
+       	wg.Wait()
		GinkgoPrint("Made %d connections in total", len(backgroundChecks))
```